File: packages/core/src/test/unit/languages/python/fixtures/local_variable_scoping.py
# Reproduces the App/Person/Student bug report for Python:
# local variables declared inside a method should never be treated as class
# fields, whether they're plain assignments or type-annotated (PEP 526).

class Person:
    def __init__(self, name):
        self.name = name


class Student(Person):
    def __init__(self, student_id, name):
        super().__init__(name)
        self.student_id = student_id


class App:
    """Well-formed, single-line method signature."""

    @staticmethod
    def main():
        s1 = Student(1, "John")
        s2: Student = Student(2, "Joe")

        print(s1 == s2)


class MultiLineSignatureApp:
    """Same scenario, but the method signature spans multiple lines -
    the Python equivalent of Java's `throws` clause breaking method-body
    detection."""

    @staticmethod
    def main(
        argv,
    ):
        s1 = Student(1, "John")
        s2: Student = Student(2, "Joe")

        print(s1 == s2)

File: packages/analysis/src/test/unit/languages/java/fixtures/LocalVariableScoping.java
package com.example.app;

/**
 * Reproduces the App/Person/Student bug report:
 * a method whose signature has a "throws" clause (so the closing
 * paren of the parameter list is NOT immediately followed by "{")
 * containing local variable declarations of a workspace class type.
 */
public class App {
    public static void main(String[] args) throws Exception {
        Student s1 = new Student(1, "John");
        Student s2 = new Student(2, "Joe");

        System.out.println(s1 == s2);
    }
}

class Person {
    private String name;

    public Person(String name) {
        this.name = name;
    }
}

class Student extends Person {
    private int studentId;

    public Student(int studentId, String name) {
        super(name);
        this.studentId = studentId;
    }
}
